### DISCUSSION TITLE
Consulta de lógica de lista simple

### DISCUSSION THREAD

**Discussion by vlzJuan** [INGCOM-UNRN-P1/cursada-2023#208] [Preguntas y respuestas] [answered]
Buenas tardes.

Consulto por el tp7: Teniendo la estructura "`nodo_t`" para hacer una lista simple, las funciones nos piden que la lista inicie como `nodo_t`. En eso, me queda un:

```
nodo_t lista = {
    dato = //entero sin usar
    next = // Puntero al elemento '0' de la lista
} (pseudo-sintaxis)
```

¿Nos importa usar ese dato de alguna forma? ¿O me estoy equivocando, y 'lista' es, en efecto, el elemento 0 de la lista?

Como consulta adicional: la función '`liberar_nodo`" nos pide como parametro un `nodo_t`, pero si es un elemento "del medio" de la lista, puede romper algo si le hago un free a secas. La idea es: Este nodo parametro, ¿Sería el 'anterior' al nodo que quiero eliminar en la lista? (En efecto, sobreescribir `lista->next` con `lista->next->next` y liberar el `lista->next` original)

**Comment by TeoInMaghine** [comment]
> ¿Nos importa usar ese dato de alguna forma? ¿O me estoy equivocando, y 'lista' es, en efecto, el elemento 0 de la lista?

Una lista sería un puntero `nodo_t`, apuntando a el primer elemento (elemento 0) de la lista, o cuando la lista está vacía apuntando a `NULL`.

> Como consulta adicional: la función 'liberar_nodo" nos pide como parametro un nodo_t, pero si es un elemento "del medio" de la lista, puede romper algo si le hago un free a secas. La idea es: Este nodo parametro, ¿Sería el 'anterior' al nodo que quiero eliminar en la lista? (En efecto, sobreescribir lista->next con lista->next->next y liberar el lista->next original)

Creo que se podría implementar como decís, siempre y cuando chequees que `lista->next` no sea `NULL` antes de hacer nada. Personalmente lo que hice fue cambiar el nombre de esa función a `liberar_lista` y que libere todos los nodos a partir del que se pasa como argumento, pero no estoy seguro cual era la intención de @martinvilu. Mi opinión es que mientras lo puedas justificar lo podrías implementar como más te sirva.

**Comment by martinvilu** [comment, marked as answer]
Pegale una leída a este hilo: https://github.com/orgs/INGCOM-UNRN-P1/discussions/200

Liberar nodo, libera el nodo individual, siempre puede existir una función que libere todos los nodos individuales hasta llegar al último, pero esto sería una función separada.

(La función de liberar nodo, puede rechazar el nodo si el siguiente no está en `NULL`, retornando un código de error.)

Lo que describís en la consulta extra, es eliminar un nodo en particular. La operación de liberación es para ser consistentes con la función de creación. La cual evitar confusiones, debe tener una función opuesta.


> ¿Nos importa usar ese dato de alguna forma? ¿O me estoy equivocando, y 'lista' es, en efecto, el elemento 0 de la lista?

Lo que vos describís, es una lista construida como la fila, del segundo punto, en donde tenés una estructura que almacena el puntero '`inicio`'
